app/services/platforms/twitch/tags.ts

```diff
@@ -0,0 +1,151 @@
+import { Observable, EMPTY } from 'rxjs';
+import { ajax } from 'rxjs/ajax';
+import { concatMap, expand, filter, map, toArray } from 'rxjs/operators';
+import { sortBy1 } from 'fp-ts/lib/Array';
+import { compose } from 'fp-ts/lib/function';
+import { fromNullable } from 'fp-ts/lib/Option';
+import { contramap, ordString } from 'fp-ts/lib/Ord';
+import { TwitchPagination } from './pagination';
+
+/**
+ * A tag on Twitch that could be assigned to a Stream.
+ */
+export type TwitchTag = {
+  tag_id: string;
+  /**
+   * `true` if the tag is autogenerated.
+   */
+  is_auto: boolean;
+  localization_names: {
+    /**
+     * Keys are locale names, in underscore (e.g. en-us). Values are the translations.
+     */
+    [x: string]: string;
+  };
+  localization_descriptions: {
+    [x: string]: string;
+  };
+};
+
+/**
+ * TwitchTag with a label in the user's locale or en-US as fallback
+ */
+export type TwitchTagWithLabel = TwitchTag & { name: string; description: string };
+
+/**
+ * Response coming from Twitch from the Tags endpoint.
+ */
+type TwitchTagsResponse = {
+  data: TwitchTag[];
+  pagination: TwitchPagination;
+};
+
+export interface TwitchRequestHeaders {
+  Accept: 'application/vnd.twitchtv.v5+json';
+  Authorization?: string;
+  'Client-Id': string;
+  'Content-Type': 'application/json';
+}
+
+/**
+ * Intermediate representation of a Twitch tag response so
+ * we can request subsequent pages of tags.
+ */
+interface PaginatedResponse {
+  items: TwitchTag[];
+  cursor: string;
+}
+
+const requestTags = (
+  headers: TwitchRequestHeaders,
+  cursor: string,
+): Observable<PaginatedResponse> =>
+  ajax
+    .getJSON<TwitchTagsResponse>(
+      `https://api.twitch.tv/helix/tags/streams?first=100&after=${cursor}`,
+      headers,
+    )
+    .pipe(
+      map(response => ({
+        cursor: response.pagination.cursor,
+        items: response.data,
+      })),
+    );
+
+/**
+ * Fetch all available tags that Twitch provides that are not
+ * automatically generated. This will use the provided pagination
+ * to request the whole dataset of tags.
+ *
+ * @param headers Headers including OAuth Token and App ID
+ */
+export const getAllTags = (headers: TwitchRequestHeaders): Promise<TwitchTag[]> =>
```

Comment:
We use `function` declarations instead of `const`. In addition to being more consistent with the current codebase, it's a more flexible way in some cases. For example, when you declare the function with `function`  keyword you don't have to care about the order of declaration.

Reply:
Hmm, do we have a strong argument against using arrow functions for this? I think not having to care about the order of declaration is actually a detriment and not a benefit, we also don't have `arguments` which is also a source of confusion sometimes. As for better stack traces, I don't believe that's true anymore, JS engines do support this case, you can check with `function.name`.